### ISSUE TITLE
Reports based on metrics

### ISSUE THREAD

**Issue by nijel** [WeblateOrg/weblate#10964]
### Describe the problem

Discussed in https://github.com/WeblateOrg/weblate/discussions/10962

<div type='discussions-op-text'>

<sup>Originally posted by **emmapeel2** February  8, 2024</sup>
I would like to easily establish how many new words we added to our website translations last year, so I can have an idea of how many updates we do.

I have been able to generate a search, and I can see we added 238 strings to this component: https://hosted.weblate.org/translate/tor/support-portal/en/?q=language%3Aen+added%3A%3E%3D2023-01-08&sort_by=-timestamp&offset=238

But I cannot find how many **words** that was.

I would like to have the possibilities of generating reports about source strings, maybe in 'insights/translation reports'.</div>

### Describe the solution you would like

The historical information is stored in the metrics, so it should be possible to expose these to the users in some way.

Presently, it is only used for trends in https://hosted.weblate.org/projects/tor/support-portal/#information

**Comment by Jibec** [comment]
A general comment on statistics:

- "string" is a technical item (useful to interact with files), useless to measure effort since a string can contain zero to dozens of words.
- "characters" is a technical item too (useful for AI costs estimation?), useless to measure effort
- "source words" sounds like the best way to measure effort to me, as it is language independent

I produce long term progress, here go Czech language:
<img width="629" height="417" alt="Image" src="https://github.com/user-attachments/assets/a3e27f6d-3f07-4a9d-94ba-46eec350a5b0" />

How do we identify what is the rework effort we have to produce to maintain software translation over time?
I feel like this kind of report align with the current request and would be most welcome in Weblate

One information that I lack is:

- how many source words changed? (measured by string updates)
- how many source words were added? (measured by string addition)
- how many source words were removed? (measure by string removal)

On the same idea, when a translation is done, how many words related to a source word addition or a source word change?

**Comment by nijel** [comment]
Words are a useful metric as long as you stick with languages that use them. 😄 

When translating from CJK languages, characters would probably be a better metric. When translating from German, the source words will mean something different from when translating from English. You're biased towards using only English as a source language, but that is not always the case.

Strings are good metrics as well in some situations, as these are the work units translators operate on. Translating 100 single-word strings is a more complex task than translating a single 100-word paragraph because you need to understand the context just once.

And those are briefly the reasons why we have all these numbers exposed in metrics.

**Comment by Jibec** [comment]
true, while hegemonic, English is not always the source language, I do have French as first language for languages-in-floss.eu content, but this is still words based

maybe a default should use words, and a configuration or whatever thing in UI allow to see the same metrics based on something else than words?
Ideally with something in URL for us to get statistics and see the use cases where people prefer to see/configure other metrics?
Maybe something in component configuration with a default to words and if language is for which word count isn't possible, suggest to use something else?

Anyway, what is new or to update remains the same, 'only' the metric change.

**Comment by emmapeel2** [comment]
Maybe the default could be changed (between words and characters) depending on the source language?
